enhancements/sig-architecture/33-hub-templates-securing-sensitive-data/README.md

```diff
@@ -0,0 +1,268 @@
+# Securing Sensitive Data in Hub Policy Templates
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in
+      [website](https://github.com/open-cluster-management/website/)
+
+## Summary
+
+When a Hub policy template is used to configure a managed cluster, the data is
+stored as plaintext in the replicated policy. This limitation prevents a hub
+cluster from securely distributing sensitive data to managed clusters using
+policies.
+
+To provide this capability, a new hub template function of `protect` will be
+added. This would encrypt the data that is piped to it using the AES-CBC
+algorithm with a 256 bit key that is unique per managed cluster. In addition, a
+`fromSecret` hub template function will be added which allows referencing data
+in a secret in the same namespace as the policy. The data from the secret would
+be automatically encrypted in the same manner as the `protect` template
+function.
+
+## Motivation
+
+When a Hub policy template is used to configure a managed cluster with sensitive
+data, it is stored in plaintext in the replicated policy. This is an issue
+because it is expected that read access to a replicated policy should not allow
+a user to see sensitive data. Additionally, when etcd encryption is enabled,
+policy objects are not encrypted. In the previous release, as a stopgap measure,
+the `fromSecret` template function was disabled to discourage storing sensitive
+data in plaintext in replicated policies. Users have requested the ability to
+configure managed clusters with sensitive data on the Hub using hub policy
+templates, so the stopgap measure is not a solution.
+
+### Goals
+
+1. A user can securely distribute sensitive data from the hub to managed
+   clusters using policies.
+1. Sensitive data can be manipulated (e.g. Base64 decode) by hub policy
+   templates before being protected in the replicated policy.
+1. The performance impact is neglible when protecting the sensitive data.
+
+### Non-Goals
+
+N/A
+
+## Proposal
+
+To allow users to configure managed clusters using policies containing arbitrary
+sensitive data, it is proposed to add a template function called `protect`,
+which will allow the user to encrypt a string in a Hub policy template. For
+example:
+
+```yaml
+'{{hub "(lookup "v1" "Secret" "default" "my-hub-secret").data.message | protect
```

Comment:
please clarify the function parameter here. if "default" is ns, why it is set here? What is allowed value, to be encrypted? can it be any interface{}?

Reply:
The namespace has to be the namespace in which the policy is created in. This is an existing restriction for Hub policy templates that does not change with this feature.

Reply:
I'll mention this in the doc for clarity.